src/classes/PMT_PaymentWizard_CTRL.cls

```diff
@@ -216,6 +205,28 @@ public with sharing class PMT_PaymentWizard_CTRL {
             writeoffPayment.npe01__Payment_Date__c = system.today();
         }
     }
+
+    /*******************************************************************************************************
+    * @description Creates an opportunity query with payment mapping and currency fields, filtered on id paramter.
+    * @return string An opportunity query string.
+    */ 
+    public string makeOppQuery() {
+        // requery the Opportunity for all fields that we need.
+        String query = 'select name, amount, stagename, npe01__payments_made__c, npe01__amount_outstanding__c, description, closedate, npe01__Number_of_Payments__c, isClosed, isWon';
+        // add any opportunity fields from payment field mappings
+        for (string s : PMT_PaymentCreator.paymentMappings.keyset()) {
+            npe01__Payment_Field_Mapping_Settings__c pfms = PMT_PaymentCreator.paymentMappings.get(s);
+            if (!query.contains(pfms.npe01__Opportunity_Field__c.toLowerCase())) {               
+                query += ', ' + pfms.npe01__Opportunity_Field__c;
+            }   
+        }       
+        //if currencyiso field exists add it to query for use later
+        if(Schema.sObjectType.Opportunity.fields.getMap().get('CurrencyIsoCode') != null)
+            query = query + ',CurrencyIsoCode';
+            
+        query += ' from Opportunity where id = :id';
```

Comment:
seems pretty obscure that this will make a query string that refers to :id that the caller had better provide.  I would have preferred seeing Id passed in, and then the value of Id put in the query string to avoid this dependency.  Especially since this is listed as a public function.  Is it really private, or used elsewhere?